internal/runtime/internal/controller/loader.go

```diff
@@ -166,7 +168,7 @@ func (l *Loader) Apply(options ApplyOptions) diag.Diagnostics {
 
 	var (
 		components   = make([]ComponentNode, 0)
-		componentIDs = make([]ComponentID, 0)
+		componentIDs = make(map[string]ComponentID)
```

Comment:
Is there a reason we switched to strings here? This feels like a regression.

Reply:
I'm using the string representation of the node ID as a key: `componentIDs[n.ID().String()] = n.ID()`
This is how the components are stored in the value_cache (this is how they were stored before as well). Using the map in the loader directly is helpful to compare the current one vs the new one in the SyncIDs function.
We could still keep it as a slice in the loader and convert it to a map in syncIDS but I thought that using a map directly would make things easier